### PR TITLE
PLAT-5870: call updateStreamInfo every 5 min

### PR DESCRIPTION
**Pull request by gadyaari**

### CODE REVIEW

File: common/config/config.json.template [hunk 1]
```diff
@@ -16,6 +16,7 @@
   "minDurationForPlaybackInSec" : 30,
   "sessionDuration" : 180000,
   "apiUpdateInterval" : 60000,
+  "streaInfoUpdateInterval" : 300000,
```
Comment (kalturaguy) [inline common/config/config.json.template]: misspling

File: lib/entry/LiveEntry.js [hunk 2]
```diff
@@ -28,13 +28,16 @@ function LiveEntry(entryObject, isNewSession) {
     this.cumulativeDurationInSec = 0;
     this.entryServerNodeId = null;
     this.entryServerType = parseInt(entryObject.serverType);
+    this.flavorsObjArray = null;
+    this.lastStreamInfoUpdate = null;
```
Comment (kalturaguy) [inline lib/entry/LiveEntry.js]: lastStreamInfoUpdate**d**

File: lib/entry/LiveEntry.js [hunk 3]
```diff
@@ -83,27 +91,24 @@ function createFlavorDownloaders(flavorsObjArray) {
...
-    return backendClient.updateStreamInfo(this.entryId, this.entryServerNodeId, flavorsObjArray)
+
+    return getEntryServerNodeId
         .then(function() {
-            that.logger.debug("[%s] Flavors updated in DB for entry server node [%s]", that.entryId, that.entryServerNodeId);
+            return backendClient.updateStreamInfo(that.entryId, that.entryServerNodeId, that.flavorsObjArray)
+                .then(function(result) {
+                    if (result) {
+                        that.logger.debug("Flavors updated in DB for entry server node [%s]", that.entryId, that.entryServerNodeId);
+                        that.lastStreamInfoUpdate = new Date().getTime();
```
Comment (kalturaguy) [inline lib/entry/LiveEntry.js]: remove getTime

File: lib/entry/LiveEntry.js [hunk 4]
```diff
@@ -45,7 +48,12 @@ function onNewTsFilesDownloaded(flavorObj, streamDurationAdditionInSec) {
         }
         else if (this.cumulativeDurationInSec >= config.get('minDurationForPlaybackInSec') && this.FSM.current !== "suspending") {
             this.logger.info("Report PLAYING to server");
-            this.FSM.play();
+            this.FSM.play()
+                .finally(function() {
+                    var currTime = new Date().getTime();
```
Comment (kalturaguy) [inline lib/entry/LiveEntry.js]: remove getTime

File: lib/BackendClient.js [hunk 5]
```diff
@@ -254,13 +254,17 @@ var BackendClient = (function(){
...
     {
         var deferred = Q.defer();
-        logger.debug("[%s] Calling updateStreamInfo for entryServerNode id [%s], with stream info array [%j]",entryId,  entryServerNodeId, liveEntryServerNode.streams);
+        logger.debug("[%s] Calling updateStreamInfo for entryServerNode id [%s], with stream info array [%j]", entryId, entryServerNodeId, liveEntryServerNode.streams);
         client.entryServerNode.update(function(result, err, headers) {
             printAPIResponse(result, headers);
             if(err) {
-                logger.debug("[%s] Failed to updateStreamInfo for entryServerNode id [%s] with error [%s]",entryId,  entryServerNodeId,  ErrorUtils.error2string(err));
+                logger.debug("[%s] Failed to updateStreamInfo for entryServerNode id [%s] with error [%s]", entryId, entryServerNodeId,  ErrorUtils.error2string(err));
+                deferred.resolve(false);
```
Comment (kalturaguy) [inline lib/BackendClient.js]: why not reject if it failed?

File: lib/entry/LiveEntry.js [hunk 6]
```diff
@@ -45,7 +47,7 @@ function onNewTsFilesDownloaded(flavorObj, streamDurationAdditionInSec) {
         }
         else if (this.cumulativeDurationInSec >= config.get('minDurationForPlaybackInSec') && this.FSM.current !== "suspending") {
             this.logger.info("Report PLAYING to server");
-            this.FSM.play();
+            this.FSM.play()
```
Comment (kalturaguy) [inline lib/entry/LiveEntry.js]: you removed the ";"

File: lib/entry/StateManager.js [hunk 7]
```diff
@@ -111,5 +115,50 @@ module.exports = function(entryObject) {
         lastLiveTimestamp = new Date();
     }
 
+    function updateStreamInfo() {
+        var curr = new Date();
+        if (curr - lastServerUpdate >= config.get('streamInfoUpdateInterval')) {
+            return getEntryServerNodeId
```
Comment (kalturaguy) [inline lib/entry/StateManager.js]: aren't we missing ()

File: lib/entry/StateManager.js [hunk 8]
```diff
@@ -87,15 +88,18 @@ module.exports = function(entryObject) {
         if ((lastServerUpdate === null) || ((currTime - lastServerUpdate) >= serverUpdateInterval)) {
             logger.info(stateMachineMessagePattern, event.toUpperCase(), from, to);
             // Refresh entry's timestamp
-            return sessionManager.refreshSessionTimestamp(entryObject.entryId)
+            return sessionManager.refreshSessionTimestamp(liveEntry.entryId)
                 .then(function() {
                     return sync.exec(function() {
-                        return backendClient.registerEntryInDatabase(entryObject, KalturaLiveStatus.PLAYABLE, event)
+                        return updateStreamInfo()
+                            .then(function() {
```
Comment (kalturaguy) [inline lib/entry/StateManager.js]: should be finally and noth then, since there could be an exception

File: lib/entry/LiveEntry.js (hunk 6 above)
Comment (gadyaari) [inline lib/entry/LiveEntry.js]: done

File: lib/BackendClient.js [hunk 9]
```diff
@@ -254,21 +254,23 @@ var BackendClient = (function(){
...
-        logger.debug("[%s] Calling updateStreamInfo for entryServerNode id [%s], with stream info array [%j]",entryId,  entryServerNodeId, liveEntryServerNode.streams);
+        logger.debug("[%s] Calling updateStreamInfo for entryServerNode id [%s], with stream info array [%j]", entryId, entryServerNodeId, liveEntryServerNode.streams);
         client.entryServerNode.update(function(result, err, headers) {
             printAPIResponse(result, headers);
             if(err) {
-                logger.debug("[%s] Failed to updateStreamInfo for entryServerNode id [%s] with error [%s]",entryId,  entryServerNodeId,  ErrorUtils.error2string(err));
+                logger.debug("[%s] Failed to updateStreamInfo for entryServerNode id [%s] with error [%s]", entryId, entryServerNodeId,  ErrorUtils.error2string(err));
+                deferred.reject();
             }
             deferred.resolve();
```
Comment (kalturaguy) [inline lib/BackendClient.js]: what if the api itself returns error?   (result is a server error)

File: lib/entry/StateManager.js [hunk 10]
```diff
@@ -111,5 +116,45 @@ module.exports = function(entryObject) {
         lastLiveTimestamp = new Date();
     }
 
+    function updateStreamInfo(currentTime) {
+        if ((currentTime - lastStreamInfoUpdate) >= config.get('streamInfoUpdateInterval')) {
```
Comment (yossipapi) [inline lib/entry/StateManager.js]: lastStreamInfoUpdateTime is more suitable

File: lib/entry/StateManager.js [hunk 11]
```diff
@@ -111,5 +116,45 @@ module.exports = function(entryObject) {
...
+            return getEntryServerNodeId()
+                .then(function () {
+                    return backendClient.updateStreamInfo(liveEntry.entryId, liveEntry.entryServerNodeId, liveEntry.flavorsObjArray)
+                        .then(function() {
+                            logger.debug("Flavors updated in DB for entry server node [%s]", liveEntry.entryServerNodeId);
+                            lastStreamInfoUpdate = new Date();
```
Comment (yossipapi) [inline lib/entry/StateManager.js]: return Q.resolve(); on success

File: lib/entry/StateManager.js (hunk 10 above)
Comment (gadyaari) [inline lib/entry/StateManager.js]: done

File: lib/entry/StateManager.js (hunk 11 above)
Comment (gadyaari) [inline lib/entry/StateManager.js]: done

File: lib/BackendClient.js (hunk 9 above)
Comment (gadyaari) [inline lib/BackendClient.js]: done

File: lib/BackendClient.js [hunk 12]
```diff
@@ -254,21 +254,23 @@ var BackendClient = (function(){
...
         var deferred = Q.defer();
-        logger.debug("[%s] Calling updateStreamInfo for entryServerNode id [%s], with stream info array [%j]",entryId,  entryServerNodeId, liveEntryServerNode.streams);
+        logger.debug("[%s] Calling updateStreamInfo for entryServerNode id [%s], with stream info array [%j]", entryId, entryServerNodeId, liveEntryServerNode.streams);
         client.entryServerNode.update(function(result, err, headers) {
             printAPIResponse(result, headers);
-            if(err) {
-                logger.debug("[%s] Failed to updateStreamInfo for entryServerNode id [%s] with error [%s]",entryId,  entryServerNodeId,  ErrorUtils.error2string(err));
+            if(err || result.objectType === 'KalturaAPIException') {
+                logger.debug("[%s] Failed to updateStreamInfo for entryServerNode id [%s] with error [%s]", entryId, entryServerNodeId,  ErrorUtils.error2string(err));
+                deferred.reject();
```
Comment (kalturaguy) [inline lib/BackendClient.js]: if you failed because of result then we won't know it

File: lib/entry/StateManager.js [hunk 13]
```diff
@@ -83,25 +85,28 @@ module.exports = function(entryObject) {
...
-                            .then(function() {
-                                lastServerUpdate = new Date().getTime();
+                        return updateStreamInfo(currTime)
+                            .finally(function() {
+                                return backendClient.registerEntryInDatabase(liveEntry.entryObject, KalturaLiveStatus.PLAYABLE, event)
+                                    .then(function() {
+                                        lastServerUpdate = new Date();
+                                    });
                             })
                             .catch(function (err) {
```
Comment (kalturaguy) [inline lib/entry/StateManager.js]: shouldn't the .catch be on the rgisterEntryInDatabasE???

File: lib/BackendClient.js (hunk 12 above)
Comment (gadyaari) [inline lib/BackendClient.js]: done

File: lib/entry/StateManager.js (hunk 13 above)
Comment (gadyaari) [inline lib/entry/StateManager.js]: done

**Pull request merged**